### PR TITLE
feat(refactor): Analysis failed jobs

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Rename analysis.failed_jobs to analysis.jobs.

### Changed

- Rename analysis.failed_jobs to analysis.jobs.

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b fix_analysis_failed_jobs -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: trailblazer/store/models.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ class Analysis(Model):
         types.Enum(*WorkflowManager.list()), default=WorkflowManager.SLURM.value
     )
 
-    failed_jobs = orm.relationship("Job", cascade="all,delete", backref="analysis")
+    jobs = orm.relationship("Job", cascade="all,delete", backref="analysis")
```
Comment (fevac) [inline trailblazer/store/models.py:81]: does this have any backward incompatibility in the database?

**Review by fevac — CHANGES_REQUESTED** [review]
OMG so nice to have this properly named! ❤️  🌟 just minor comments on my side. Nicely done!

File: trailblazer/server/api.py [hunk 2]
```diff
@@ -59,7 +63,7 @@ def analyses():
     for analysis_obj in query_page.items:
         analysis_data = analysis_obj.to_dict()
         analysis_data["user"] = analysis_obj.user.to_dict() if analysis_obj.user else None
-        analysis_data["failed_jobs"] = [job_obj.to_dict() for job_obj in analysis_obj.failed_jobs]
+        analysis_data["failed_jobs"] = [job_obj.to_dict() for job_obj in analysis_obj.jobs]
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply trailblazer/server/api.py:66]: Changing key would break the API. It deserves its own PR.

File: trailblazer/store/models.py (hunk 1 above)
Comment (henrikstranneheim) [reply trailblazer/store/models.py:81]: No it should not

File: trailblazer/store/api.py [hunk 3]
```diff
@@ -318,7 +322,7 @@ def cancel_analysis(self, analysis_id: int, email: str = None, ssh: bool = False
         if analysis.status not in TrailblazerStatus.ongoing_statuses():
             raise TrailblazerError(f"Analysis {analysis_id} is not running")
 
-        for job_obj in analysis.failed_jobs:
+        for job_obj in analysis.jobs:
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply trailblazer/store/api.py]: Fixed

File: tests/cli/test_cli_core.py [hunk 4]
```diff
@@ -138,7 +138,7 @@ def test_cancel_ongoing(cli_runner, trailblazer_context, caplog):
         )
 
         # Analysis should have jobs that can be cancelled
-        assert analysis_obj.failed_jobs
+        assert analysis_obj.jobs
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply tests/cli/test_cli_core.py]: Fixed

**Review by fevac — APPROVED** [review]
🚀

**Comment by henrikstranneheim** [comment]
Redid the same tests - all pass

**Comment by henrikstranneheim** [comment]
Deployed in production:

```
Log deploy... done.
trailblazer, version 16.0.3
θ76° [henrik.stranneheim@hasta:~] [P_base] 1m36s $
```

**Pull request merged**